### PR TITLE
*: apply Go 1.19

### PR DESCRIPTION
**Pull request by 53jk1**

### CODE REVIEW

**Comment by nicktrav** [comment]
Thanks for the patch.

Can you rebase to pick up the changes from #2250. That will allow this PR to build correctly.

File: go.mod [hunk 1]
```diff
@@ -1,43 +1,44 @@
 module github.com/cockroachdb/pebble
 
 require (
-	github.com/DataDog/zstd v1.4.5
+	github.com/DataDog/zstd v1.5.2
```
Comment (nicktrav) [inline go.mod:4]: I don't think we want all of these version bumps in the same PR. Those would ideally be handled separately. For some of the dependencies (`xxhash`, `compress`, etc.), we'll also want to run some benchmarks. There have been some issues in the past.

Can we keep this PR focused on bumping the build version to `1.19` (i.e. should be a single line change)?

Comment (53jk1) [reply go.mod:4]: Yes, we can keep this PR focused on bumping the build version to 1.19. I'll create separate PRs for the other version bumps and will make sure to run the necessary benchmarks before submitting them. Thanks for the suggestion.

Comment (53jk1) [reply go.mod]: @nicktrav you can check this PR again, I bumped only the build version to `1.19`. Other than that, I have run the Makefile locally and everything looks fine.

**Review by nicktrav — APPROVED** [review]
:lgtm: - cc: @jbowens 

> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/pebble/2246#-:-NMW3BPz9D7hEvDiw87-:bxipafi)* status: all files reviewed, 1 unresolved discussion (waiting on @53jk1)

**Review by jbowens — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/pebble/2246#-:-NMZG-Nh0Fhi5WSzaDFk:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @nicktrav)

**Pull request merged**